bluetooth: controller: Disable Coded PHY by default

rugeGerritsen opened the pull request:
Coded PHY is not needed in most applications.
By disabling Coded PHY, most applications will:
 - Get smaller in terms of flash and ram usage
 - Not get vurnerable to devices trying to switch to Coded and
   thereby increasing power consumption.

HCI samples have Coded PHY enabled by default, as they expect all
features to be enabled.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

joerchan reviewed (commented):
Ok with me, but should me mentioned in the release notes that the default has changed.

cvinayak commented on samples/bluetooth/hci_rpmsg/prj.conf:
```diff
@@ -16,3 +16,5 @@ CONFIG_BT_HCI_RAW_RESERVE=1
 CONFIG_ASSERT=y
 CONFIG_DEBUG_INFO=y
 CONFIG_EXCEPTION_STACK_TRACE=y
+
+CONFIG_BT_CTLR_PHY_CODED=y
```
```suggestion

```

cvinayak commented on samples/bluetooth/hci_spi/prj.conf:
```diff
@@ -8,3 +8,5 @@ CONFIG_BT_MAX_CONN=20
 CONFIG_BT_TINYCRYPT_ECC=n
 CONFIG_BT_RX_BUF_COUNT=10
 CONFIG_BT_RX_BUF_LEN=255
+
+CONFIG_BT_CTLR_PHY_CODED=y
```
```suggestion
```

cvinayak commented on samples/bluetooth/hci_uart/prj.conf:
```diff
@@ -14,6 +14,7 @@ CONFIG_BT_CTLR_ASSERT_HANDLER=y
 CONFIG_BT_MAX_CONN=16
 CONFIG_BT_TINYCRYPT_ECC=n
 CONFIG_BT_CTLR_DTM_HCI=y
+CONFIG_BT_CTLR_PHY_CODED=y
```
```suggestion
```

cvinayak commented on samples/bluetooth/hci_usb/prj.conf:
```diff
@@ -10,3 +10,5 @@ CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
 CONFIG_USB_DEVICE_BLUETOOTH=y
 CONFIG_USB_DEVICE_BLUETOOTH_VS_H4=y
+
+CONFIG_BT_CTLR_PHY_CODED=y
```
```suggestion
```

cvinayak commented on samples/bluetooth/hci_usb_h4/prj.conf:
```diff
@@ -6,3 +6,5 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
 CONFIG_USB_DEVICE_BT_H4=y
+
+CONFIG_BT_CTLR_PHY_CODED=y
```
```suggestion
```

cvinayak commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -430,7 +430,6 @@ config BT_CTLR_PHY_2M
 config BT_CTLR_PHY_CODED
 	bool "Coded PHY Support"
 	depends on (BT_CTLR_PHY || BT_CTLR_ADV_EXT) && BT_CTLR_PHY_CODED_SUPPORT
```
```suggestion
	depends on (BT_CTLR_PHY || BT_CTLR_ADV_EXT) && BT_CTLR_PHY_CODED_SUPPORT
	default y if BT_HCI_RAW
```

cvinayak reviewed (changes requested):
Add `default y if BT_HCI_RAW` to Kconfig instead of adding `CONFIG_BT_CTLR_PHY_CODED=y` to prj.conf files.

rugeGerritsen committed:
bluetooth: controller: Disable Coded PHY by default.

Coded PHY is only enabled by default if the application is using
BT_HCI_RAW.

Coded PHY is not needed in most applications.
By disabling Coded PHY, most applications will:

* Get smaller in terms of flash and ram usage
* Not get vurnerable to devices trying to switch to Coded and
  thereby increasing power consumption.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

joerchan commented on doc/releases/release-notes-2.5.rst:
```diff
@@ -62,6 +62,8 @@ API Changes
   parameter, allowing to specify the maximum retransmission count of the
   confirmable message.
 
+* The ``CONFIG_BT_CTLR_CODED_PHY`` is now disabled by default.
```
in combined builds. Since you changed it.
